File: Brute Force/stack.cpp
#include<iostream>
using namespace std;
#include<stack>

int main()
{  int n,k=0;
int value=-1;
cin>>n;
    int a[n],a2[n];
    for(int i=0;i<n;i++)
    {
        cin>>a[i];
    }

    stack<int> mystack;
    for(int i=0;i<n;i++)
    {  int j;
        for(j=i+1;j<n;j++)
        {
            if(a[j]>a[i])
            {
                mystack.push(a[j]);
                a2[k]=mystack.top();
                k++;
                break;
            }

        }
          if(a[j]<a[i])
            {mystack.push(0);
            a2[k]=mystack.top();
            k++;}

    }


    for(int i=0;i<n;i++)
    {
        cout<<a2[i];
    }
}

File: Stack/JavaScript/stack.js
//using Linked List
class Node {
  constructor(value){
    this.value = value;
    this.next = null;
  }
}

class Stack {
  constructor(){
    this.top = null;
    this.bottom = null;
    this.length = 0;
  }
  isEmpty(){
      return this.length === 0;
  }
  peek() {
      if(this.isEmpty()){
          console.log('Stack is empty!')
      }else{
          console.log(this.top.value);
      }
  }
  push(value){
      let currNode = new Node;
      currNode.value = value;
      if(this.isEmpty()){
          this.top = currNode;
          this.bottom = currNode;
          
      }
      else{
          this.top.next = currNode;
          this.top = currNode;
          
      }
      this.length++;
  }
  pop(){
      if(this.isEmpty()) return null;
      let currNode = this.top ;
      let pointer = this.bottom;
      while(pointer.next && pointer.next !=this.top){
        pointer = pointer.next;
      }
      currNode = null;
      this.top = pointer;
      this.length--;
  }
  //isEmpty
}

const myStack = new Stack();

myStack.push(4);
myStack.push(6);
myStack.push(8);
myStack.push(10)
myStack.push(12);
myStack.push(13);
myStack.pop();
myStack.peek();
myStack.pop();
myStack.peek();
myStack.pop();
myStack.peek();
myStack.pop();
myStack.peek();
myStack.pop();
myStack.peek();
myStack.pop();
myStack.peek();
